apps/aevm/src/aevm_eeevm_store.erl

```diff
@@ -196,31 +195,33 @@ store_maps(Maps0, Store) ->
 perform_update({new_inplace, NewId, OldId}, Store) ->
     OldKey   = <<OldId:256>>,
     NewKey   = <<NewId:256>>,
-    OldEntry = maps:get(OldKey, Store),
-    Store1   = maps:remove(OldKey, Store),
-    Store1#{ NewKey => OldEntry };
+    OldEntry = store_get(OldKey, Store),
+    %% Subtle: Don't remove the old entry because the mp trees requires there
```

Comment:
This is problematic. And I guess this is the kicker for why you test the content rather than the MPT above.

If you leave old values lying around, the root hash of the store changes, as will the root hash of the state. This is not good enough since it is under consensus.

Reply:
I don't think there's a problem leaving the old entry around (but, yes this does affect consensus). This is just an internal detail in how contracts manage the store and shouldn't affect things outside.